stats/stats_test.go

```diff
@@ -940,17 +940,21 @@ func TestClientStatsUnaryRPCError(t *testing.T) {
 		failfast:   failfast,
 	}
 
-	checkFuncs := []func(t *testing.T, d *gotData, e *expectedData){
-		checkBegin,
-		checkOutHeader,
-		checkOutPayload,
-		checkInHeader,
-		checkInTrailer,
-		checkEnd,
+	checkFuncs := map[int]*checkFuncWithCount{
```

Comment:
You'd need to check for nil anyway, since the map keys give you no guarantee whatsoever.